src/Proto3/Wire/Decode.hs

```diff
@@ -215,21 +220,18 @@ takeWT LengthDelimited b = do
 -- * Parser Interface
 
 -- | Type describing possible errors that can be encountered while parsing.
-data ParseError =
-                -- | A 'WireTypeError' occurs when the type of the data in the protobuf
-                -- binary format does not match the type encountered by the parser. This can
-                -- indicate that the type of a field has changed or is incorrect.
-                WireTypeError Text
-                |
-                -- | A 'BinaryError' occurs when we can't successfully parse the contents of
-                -- the field.
-                BinaryError Text
-                |
-                -- | An 'EmbeddedError' occurs when we encounter an error while parsing an
-                -- embedded message.
-                EmbeddedError Text
-                              (Maybe ParseError)
-    deriving (Show, Eq, Ord)
+data ParseError
+  = -- | A 'WireTypeError' occurs when the type of the data in the protobuf
```

Comment:
As @evanrelf  mentioned on another PR, it might be helpful to have a separate PR for widespread style changes to code that you are not otherwise touching.  I don't mind a bit of unrelated change, but this is pretty massive.